### PR TITLE
Separate cavy-cli reporter

### PR DESCRIPTION
**Pull request by AbigailMcP**
**Includes**
- Refactoring cavy-cli into a reporter function (this has the side benefit of resembling the custom reporters we are soon going to allow users to pass in)
- Allow custom reporter function to be passed into tester component and used (if defined) instead of cavy-cli

### CODE REVIEW

**Comment by jalada** [comment]
Note for merging: Patch version bump.

File: src/TestRunner.js [hunk 1]
```diff
@@ -11,10 +11,11 @@
 //              tests.
 export default class TestRunner {
 
-  constructor(component, testSuites, startDelay, sendReport) {
+  constructor(component, testSuites, startDelay, reporter, sendReport) {
```
Comment (jalada) [inline src/TestRunner.js]: Are we OK with how long this is getting?

File: src/TestRunner.js [hunk 2]
```diff
@@ -11,10 +11,11 @@
...
     this.component = component;
     this.testSuites = testSuites;
     this.startDelay = startDelay;
+    this.sendReport = reporter;
```
Comment (jalada) [inline src/TestRunner.js]: `this.sendReport` vs `sendReport` is a little confusing; this looks like a typo at first. Could this just be `this.reporter`?

File: src/reporter.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    method: 'POST',
+    body: JSON.stringify(report),
+    headers: { 'Content-Type': 'application/json' }
+  };
+
+  try {
+    await fetch(url, options);
+    console.log('Cavy test report successfully sent to cavy-cli');
+  } catch (e) {
+    if (e.message.match(/Network request failed/)) {
```
Comment (jalada) [inline src/reporter.js]: As per comment in other PR

File: src/TestRunner.js (hunk 1 above)
Comment (AbigailMcP) [reply src/TestRunner.js]: @jalada - would you prefer an options object?

**Comment by AbigailMcP** [comment]
> Note for merging: Patch version bump.

This now includes the ability to pass in your own reporter -> Minor bump

**Pull request merged**